code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,223 @@
+				value = appearance.mouse_opacity
+			if("name")
+				value = appearance.name
+			if("opacity")
+				value = appearance.opacity
+			if("overlays")
+				value = appearance.overlays
+			if("override")
+				value = appearance.override
+			if("pixel_x")
+				value = appearance.pixel_x
+			if("pixel_y")
+				value = appearance.pixel_y
+			if("pixel_w")
+				value = appearance.pixel_w
+			if("pixel_z")
+				value = appearance.pixel_z
+			if("plane")
+				value = appearance.plane
+			if("render_source")
+				value = appearance.render_source
+			if("render_target")
+				value = appearance.render_target
+			if("suffix")
+				value = appearance.suffix
+			if("text")
+				value = appearance.text
+			if("transform")
+				value = appearance.transform
+			if("underlays")
+				value = appearance.underlays
+
+			if("parent_type")
+				value = appearance.parent_type
+			if("type")
+				value = appearance.type
+
+			// These are not documented ones but trackable values. Maybe we'd want these.
+			if("animate_movement")
+				value = atom_appearance.animate_movement
+			if("dir")
+				value = atom_appearance.dir
+			if("glide_size")
+				value = atom_appearance.glide_size
+			if("pixel_step_size")
+				value = "" //atom_appearance.pixel_step_size
+				// DM compiler complains this
+
+			// These variables are only available in some conditions.
+			if("contents")
+				value = atom_appearance.contents
+			if("vis_contents")
+				value = atom_appearance.vis_contents
+			if("vis_flags")
+				value = atom_appearance.vis_flags
+			if("loc")
+				value = atom_appearance.loc
+			if("locs")
+				value = atom_appearance.locs
+			if("x")
+				value = atom_appearance.x
+			if("y")
+				value = atom_appearance.y
+			if("z")
+				value = atom_appearance.z
+
+			// we wouldn't need these, but let's these trackable anyway...
+			if("cooldowns")
+				value = atom_appearance.cooldowns
+			if("gc_destroyed")
+				value = atom_appearance.gc_destroyed
+			if("datum_components")
+				value = atom_appearance.datum_components
+			if("datum_flags")
+				value = atom_appearance.datum_flags
+			if("density")
+				value = atom_appearance.density
+			if("screen_loc")
+				value = atom_appearance.screen_loc
+			if("sorted_verbs")
+				value = atom_appearance.sorted_verbs
+			if("tag")
+				value = atom_appearance.tag
+			if("tgui_shared_states")
+				value = atom_appearance.tgui_shared_states
+			if("verbs")
+				value = atom_appearance.verbs
+			if("weak_reference")
+				value = atom_appearance.weak_reference
+			if("cached_ref")
+				value = appearance.cached_ref
+
+			else
+				return "<li style='backgroundColor:white'>(STATIC) [var_name] <font color='blue'>(Undefined var name in switch)</font></li>"
+	catch
+		return "<li style='backgroundColor:white'>(STATIC) <font color='blue'>[var_name] = (untrackable)</font></li>"
+	return "<li style='backgroundColor:white'>(STATIC) [var_name] = [_debug_variable_value(var_name, value, 0, appearance, sanitize = TRUE, display_flags = NONE)]</li>"
```

Comment:
Rather than static I would put something like (READ ONLY) since I get the intention but it isn't static (static would be shared across all instances of /appearance, whereas appearances are more like singleton instances). Read only would be more proper terminology I think